src/Types/ListType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace ILIAS\Types;
+
+/**
+ * Class ListType
+ *
+ * @package ILIAS\Types
+ *
+ * @author  Oskar Truffer <ot@studer-raimann.ch>
+ *
+ * Describes a list of a certain Type.
+ *
+ * Attention:
+ * isExtensionOf behaves Covariant!
+ * new ListType(Dog::class).isExtensionOf(new ListType(Animal::class)) == true!
+ *
+ * See: https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)
+ */
+class ListType implements Type, Ancestors {
+
+	/** @var Type */
+	protected $type;
+
+
+	/**
+	 * SingleType constructor.
+	 *
+	 * @param $fullyQualifiedClassName string|Type Give a Value Type or a Type that will be wrapped in a single type.
+	 */
+	public function __construct($fullyQualifiedClassName) {
+		if (!is_a($fullyQualifiedClassName, Type::class)) {
+			$fullyQualifiedClassName = new SingleType($fullyQualifiedClassName);
+		}
+		$this->type = $fullyQualifiedClassName;
+	}
+
+
+	/**
+	 * @inheritdoc
+	 */
+	function __toString() {
+		return "[" . $this->type . "]";
+	}
+
+
+	/**
+	 * Is this type a subtype of $type. Not strict! x->isExtensionOf(x) == true.
+	 *
+	 * Attention: [Dog].isExtensionOf([Animal]) == true. In other words:
+	 * The isExtensionOf behaves Covariant!
+	 *
+	 * If you are familiar with e.g. Java Generics ?.isExtensionOf(x) behaves the same as <? extends x>.
```

Comment:
Hi Oskar,

I don't see how this comment solves this problem in any way. `SingleType::isExtensionOf` implements this method by using `isSubclassOf`, which conveys a different meaning than the one announced here. This implementation is also different from the interface defined in `Type`. There is no meaningful way to implement a `List of (subclass of X)s` as a subclass of `List of Xs`. The link in the comment explains that properly. I thus don't see which information I (as a user of Type) get, if I call `isExtensionOf` on a type object.

Best regards!